### PR TITLE
feat:exposed visibility value for the provider in the DA

### PR DESCRIPTION
**Pull request by Aditya-ranjan-16**
### Description

exposed [visibility](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs#visibility) value for the provider in the DAs , with default value set to "private"

[Git issue](https://github.ibm.com/GoldenEye/issues/issues/11337)

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

exposes visibility value for the provider in the DA

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: ibm_catalog.json [hunk 1]
```diff
@@ -118,7 +118,26 @@
                 "description": "This deployable architecture creates IBM Cloud Code Engine and configures the project, apps, domain mappings, secrets, configmaps, and service bindings."
               }
             ]
+          },
```
Comment (akocbek) [inline ibm_catalog.json:121]: noticed we do not have all input DA variables defined inside ibm-catalog.json configuration file.
we should sync with other DAs and please add other input variables to `configuration` key inside.

for example redis has all inputs: https://github.com/terraform-ibm-modules/terraform-ibm-icd-redis/blob/7731fa1aebc896c658e034b3a3728d64443cf9f0/ibm_catalog.json#L213

File: ibm_catalog.json [hunk 2]
```diff
@@ -157,7 +176,26 @@
                     "description": "This deployable architecture creates IBM Cloud Code Engine projects"
                 }
                 ]
+            },
```
Comment (akocbek) [inline ibm_catalog.json:179]: same here, we need all input variables

**Review by akocbek — CHANGES_REQUESTED** [review]

File: ibm_catalog.json (hunk 2 above)
Comment (Aditya-ranjan-16) [reply ibm_catalog.json:310]: added https://github.com/terraform-ibm-modules/terraform-ibm-code-engine/pull/132/commits/455789f8e6a4428bbda8cfc4a30d9124498d6318

File: ibm_catalog.json [hunk 3]
```diff
@@ -157,7 +307,84 @@
...
+            {
+              "key": "prefix"
+            },
+            {
+              "key": "existing_resource_group"
+            },
+            {
+              "key": "resource_group_name"
+            },
+            {
```
Comment (ocofaigh) [inline ibm_catalog.json:341]: same [comment](https://github.com/terraform-ibm-modules/terraform-ibm-code-engine/pull/132/files#r1857340740) as above

File: ibm_catalog.json [hunk 4]
```diff
@@ -118,7 +118,157 @@
...
+          },
+          {
+            "key": "resource_group_name"
+          },
+
+          {
+            "key": "region",
+            "required": true,
+            "default_value": "us-south",
+            "options": [
```
[the remark this answers is not in the document]
Comment (Aditya-ranjan-16) [reply ibm_catalog.json]: done https://github.com/terraform-ibm-modules/terraform-ibm-code-engine/pull/132/commits/25cda97e4aa758d74d1145f89458e2431242cfe5

File: ibm_catalog.json (hunk 3 above)
Comment (Aditya-ranjan-16) [reply ibm_catalog.json:311]: done https://github.com/terraform-ibm-modules/terraform-ibm-code-engine/pull/132/commits/25cda97e4aa758d74d1145f89458e2431242cfe5

**Review by shemau — COMMENTED** [review]
This is neither an approval or a request for changes. The original scope of the issue is growing, so should this be split to a new issue or resolved in this issue?

The description of variables in the apps solution needs some work, and/or a mark down written to explain the 6 compelx variables types in the DA.

There are examples of mark down, eg. ICD Postgresql in solution/standard/DA-types.md.

The lists and maps are fundamentally different things. The lists specifically belong to the app. The maps are other code engine resources. So there may be a hybrid answer from a DA perspective, it may be better to remove the maps from the apps DA and to include new DAs to cover them.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Please always explain why upgrade test was skipped. I see no reason why a change like this should need an upgrade test to be skipped.

File: ibm_catalog.json [hunk 5]
```diff
@@ -118,7 +118,127 @@
...
+          },
+          {
+            "key": "image_secret"
+          },
+          {
+            "key": "run_env_variables"
+          },
+          {
+            "key": "run_volume_mounts"
+          },
```
[the remark this answers is not in the document]
Comment (ocofaigh) [reply ibm_catalog.json:183]: @shemau we already have a tracking issue for this: https://github.com/terraform-ibm-modules/terraform-ibm-code-engine/issues/63

File: ibm_catalog.json [hunk 6]
```diff
@@ -118,7 +118,127 @@
...
+            "key":"config_maps"
+          },
+          {
+            "key":"secrets"
+          },
+          {
+            "key":"domain_mappings"
+          },
+          {
+            "key":"bindings"
```
[the remark this answers is not in the document]
Comment (ocofaigh) [reply ibm_catalog.json:239]: we already have a tracking issue for this: https://github.com/terraform-ibm-modules/terraform-ibm-code-engine/issues/63

**Comment by ocofaigh** [comment]
@shemau There is an issue to write up some supporting docs for the object type inputs: https://github.com/terraform-ibm-modules/terraform-ibm-code-engine/issues/63

I'm going to proceed with merging this PR

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**